Assets/MRTK/SDK/Experimental/SpatialAwareness/SurfaceMeshesToPlanes.cs

```diff
@@ -0,0 +1,464 @@
+        {
+            Destroy(PlanesParent);
+            tokenSource?.Cancel();
+        }
+
+        /// <summary>
+        /// Task to analyze surface meshes to find planes and create new 3D cubes to represent each plane.
+        /// </summary>
+        /// <param name="cancellationToken">Token that allows cancellation of an asynchronous process.</param>
+        /// <returns>Yield result.</returns>
+        private async Task MakePlanes(CancellationToken cancellationToken)
+        {
+            await new WaitForUpdate();
+
+#if PLANE_FINDING_PRESENT
+            List<PlaneFinding.MeshData> meshData = new List<PlaneFinding.MeshData>();
+            List<MeshFilter> filters = new List<MeshFilter>();
+
+            var spatialAwarenessSystem = CoreServices.SpatialAwarenessSystem;
+            if (spatialAwarenessSystem != null)
+            {
+                GameObject parentObject = spatialAwarenessSystem.SpatialAwarenessObjectParent;
+
+                // Get mesh filters from SpatialAwareness Mesh Observer
+                foreach (MeshFilter filter in parentObject.GetComponentsInChildren<MeshFilter>())
+                {
+                    filters.Add(filter);
+                }
+            }
+
+            for (int index = 0; index < filters.Count; index++)
+            {
+                MeshFilter filter = filters[index];
+                if (filter != null && filter.sharedMesh != null)
+                {
+                    // fix surface mesh normals so we can get correct plane orientation.
+                    filter.mesh.RecalculateNormals();
+                    meshData.Add(new PlaneFinding.MeshData(filter));
+                }
+            }
+
+            BoundedPlane[] planes;
+
+            await new WaitForBackgroundThread();
+            {
+                planes = PlaneFinding.FindPlanes(meshData, SnapToGravityThreshold, MinArea);
+            }
+
+            await new WaitForUpdate();
+
+            DestroyPreviousPlanes();
+            activePlanes.Clear();
+            ClassifyAndCreatePlanes(planes);
+
+            // We are done creating planes, trigger an event.
+            MakePlanesComplete?.Invoke(this, EventArgs.Empty);
+#endif // PLANE_FINDING_PRESENT
+            makingPlanes = false;
+        }
+
+#if PLANE_FINDING_PRESENT
+        /// <summary>
+        /// Create game objects to represent bounded planes in scene
+        /// </summary>
+        private void ClassifyAndCreatePlanes(BoundedPlane[] planes)
+        {
+            SetFloorAndCeilingPositions(planes);
+
+            // Create SurfacePlane objects to represent each plane found in the Spatial Mapping mesh.
+            for (int index = 0; index < planes.Length; index++)
+            {
+                BoundedPlane boundedPlane = planes[index];
+
+                var planeObject = SpatialAwarenessPlanarObject.CreateSpatialObject(
+                    boundedPlane.Bounds.Center,
+                    boundedPlane.Bounds.Extents,
```

Comment:
Can extents param of `CreateSpatialObject` be changed back to the original parameter of `size` and we pass `PlaneThickness` as the z value? that way we don't have to manually set it below on line 335?

Reply:
Though, i guess this was is an "either way" thing. I could see us only wanting extents and providing a standard thickness in the implementation

Reply:
sure!